There is a logic issue which prevents the gem server from running.

Patch from upstream:
https://github.com/ruby/rdoc/commit/82d59dd0369155153701d96aa8c86cb77780b5ed
https://github.com/ruby/rdoc/commit/7f8582da455a925979da739078d6dd5df2a8956d

--- ruby-2.5.5/lib/rdoc/servlet.rb
+++ ruby-2.5.5/lib/rdoc/servlet.rb
@@ -111,7 +111,7 @@ class RDoc::Servlet < WEBrick::HTTPServl
   # GET request entry point.  Fills in +res+ for the path, etc. in +req+.
 
   def do_GET req, res
-    req.path = req.path.sub(/^#{Regexp.escape @mount_path}/o, '') if @mount_path
+    req.path.sub!(/^#{Regexp.escape @mount_path}/o, '') if @mount_path
 
     case req.path
     when '/' then
--- ruby-2.5.5/test/rdoc/test_rdoc_servlet.rb
+++ ruby-2.5.5/test/rdoc/test_rdoc_servlet.rb
@@ -143,7 +143,7 @@ class TestRDocServlet < RDoc::TestCase
 
       @s.asset_dirs[:darkfish] = '.'
 
-      @req.path = '/mount/path/css/rdoc.css'
+      @req.path = '/mount/path/css/rdoc.css'.dup
 
       @s.do_GET @req, @res
 
